internal/service/employeeaccessgrant/model.go

```diff
@@ -0,0 +1,30 @@
+package employeeaccessgrant
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/conversion"
+	"go.mongodb.org/atlas-sdk/v20240805003/admin"
+)
+
+func NewTFModel(projectID, clusterName string, apiResp *admin.EmployeeAccessGrant) *TFModel {
```

Comment:
I guess these functions will never be called with `nil`. However, uncertain if it makes sense to add this check :/

Reply:
not a strong opinion, I'm following a "minimalist/TDD-ish" approach of implementing only what is needed.  
In this case, at the moment, it's an error to pass a nil param so I'm ok with panic.  
As an example it should NOT return nil if param is nil, that would because a logical error somewhere so better to notice it than to "hide" the error and do something unexpected